lib/Pipes/RootKind.cpp

```diff
@@ -62,3 +62,11 @@ void RootKind::getInvalidations(const Context &Ctx,
   if (RootChanged)
     ToRemove.emplace_back(*this);
 }
+
+void RootKind::allTargets(const Context &Ctx, TargetsList &Out) const {
+  Out.push_back(Target(*this));
+}
+
+void IsolatedRootKind::allTargets(const Context &Ctx, TargetsList &Out) const {
+  Out.push_back(Target(*this));
+}
```

Comment:
Isn't this the default implementation of `UniqueKind`? Can it be?

Reply:
unfortunately the other virtual function, getInvalidations, will be customized by the invalidation mr

Reply:
You seem to assume the invalidations PR will go in first. This PR seems quite ready to go in to me.

In general, I didn't understand your answer. If these are `UniqueKind`, we can make them have this as the default implementation and then let people override it if needed.

Reply:
ahh, i thought you wanted to replace RootKind with UniqueKind entirelly. What you propose works.

Reply:
RootKind extends LLVMKind, which is unrelated to UniqueKind. we can introduce a UniqueLLVMKind that implements allTargets, but this two would be the only two users. I don't see great utility in having them.